accepted/0000-timezones.md

```diff
@@ -0,0 +1,103 @@
+- Feature Name: timezones
+- Start Date: 2019-09-16
+- RFC PR: 
+
+# Summary
+[summary]: #summary
+
+Timezones can be a big problem when managing an application like Uyuni that can be deployed and accessed from anywhere in the world. 
+If we don’t have some rules on how to display and handle dates, this can become quite a big mess to manage. 
+
+
+# Motivation
+[motivation]: #motivation
+
+The goal of this RFC is to define those rules and serve as a reference for the future.
+
+These rules should define a consistent way of display dates on the UI and make this information as clear as possible for the user.
+
+The user should be able to easily understand in which timezone a date is displayed and aware if there is any difference from the timezone
+he is accessing from.
+
+# Glossary
+[glossary]: #glossary
+
+When talking about uyuni there are 4 different Timezones contexts that are important to understand and distinguish:
+
+* **Managed systems timezone:** the timezone of each minion/client. This doesn't have any impact on this RFC, but it's important to mention.
+
+* **Uyuni Server timezone:** the configured timezone in the server where uyuni is deployed 
+
+* **User preferences timezone:** the defined timezone on the user preferences settings. Here we should be able to choose the timezone, date format (yyyyddmm) and the time format (12/24 hours).
+All the shown dates on the UI and operations on them should always assume these user settings. 
+
+* **Client browser timezone:** the browser/pc timezone where the client is accessing from. This timezone shouldn’t affect any displayed date, but the user should be aware of it.
+
+
+# Detailed design
+[design]: #detailed-design
+
+There are 3 topics that we need to have in consideration here:
+
+## Display of dates
+
+The display of dates should always take in consideration the 3 user preferences settings: timezone, date-format and time-format.
+
+**For example:** '2019-09-10 13:00:00 **CEST**'
+
+At the moment only the setting timezone is implemented. The property date-format and time-format could be added on a second iteration, 
+but the code should be prepared to easily support them.
+
+Example:
+
+![timezones](images/timezones2.png)
+
+Ideally, these dates should be formatted server side using the already existing java methods:
+
+```java
+LocalizationService.getInstance().formatCustomDate(date)
+```
+
+For the new stack (spark+react) if you want to convert a JSON to use the correct format, use the following GSON date adapter `UserLocalizationDateAdapter` 
+that will convert your class dates to the user defined format.
+
+The previous gson adapter 'ECMAScriptDateAdapter' that converts dates to UTC should be avoided.
+
+## Manipulation of dates on the browser
+
+Since the browser Date implementation doesn't support timezones, if we ever need to do any date manipulation more than displaying a date
+ we should never use the object Date() directly but the `moment-timezone` library.
```

Comment:
Will we package it or vendorize it?

Reply:
We will vendorize it in the same we already do for all javascript libraries. 

We download all the libraries in jenkins (yarn install) -> generate a zip and licenses -> upload it to OBS -> generate js bundles in OBS -> susemager-web-libs.rpm (sub module of spacewalk-web)

There is already an open PR to add this library: https://github.com/uyuni-project/uyuni/pull/1408